### PR TITLE
Next

### PR DESCRIPTION
**Pull request by alex13bright**

### CODE REVIEW

File: app/components/common/ScrollToTop.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { ReactElement, ReactNode, useLayoutEffect } from 'react'
+import { useLocation } from 'react-router-dom'
+
+type Props = {
+  children: ReactNode
+}
+export const ScrollToTop = ({ children }: Props): ReactElement => {
```
Comment (alex-adamant) [inline app/components/common/ScrollToTop.tsx:7]: It doesn't need to be a component, does it? Just a hook will do, as it seems.

File: app/lib/utilities.ts [hunk 2]
```diff
@@ -70,3 +70,7 @@ export const toObject = <K extends string, O extends { [k in K]: string }>(
   arr: O[],
   k: K
 ): { [k in O[K]]: Omit<O, K> } => fromEntries(arr.map(({ [k]: v, ...p }) => [v, p]))
+
+export const never = (_: never): never => {
```
Comment (alex-adamant) [inline app/lib/utilities.ts:74]: A little bit confusing name for the function. I'd either add a verb or at least make it different from `ts` keyword.

**Review by alex-adamant — COMMENTED** [review]
Totally realising that it's a draft, I just quickly went through the main changes and dropped a couple of ideas.

File: app/components/common/ScrollToTop.tsx (hunk 1 above)
Comment (alex-adamant) [reply app/components/common/ScrollToTop.tsx:7]: Unless you're going to render it conditionally, but then I might have not fully understood the use case for it.

Comment (alex13bright) [reply app/components/common/ScrollToTop.tsx:7]: It's taken from SO. Should be a hook for sure. Thanks!

File: app/lib/utilities.ts (hunk 2 above)
Comment (alex13bright) [reply app/lib/utilities.ts:74]: Renamed.

File: app/components/common/ScrollToTop.tsx (hunk 1 above)
Comment (alex13bright) [reply app/components/common/ScrollToTop.tsx:7]: By the way it does not work. Except using plain "a" tag for full page reloading how to scroll page to top on navigation with js?

Comment (alex-adamant) [reply app/components/common/ScrollToTop.tsx]: On my pre-previous job I did it like this:

```
export default function ScrollRestorationProvider(props: ParentComponentProps) {
  const scrollRef = useRef<HTMLDivElement | null>(null)
  const existingContext = useContext(ScrollRestorationContext)

  const { state } = useLocation()
  const { scrollTop = 0 } = state || ({} as any)

  useLayoutEffect(() => {
    if (existingContext) return
    // For some unclear at the moment reasons, this effect fires before full children repaint, so it won't work without setTimeout
    setTimeout(() => {
      if (scrollTop && scrollRef.current) {
        scrollRef.current.scrollTop = scrollTop
      }
    }, 800)
  }, [existingContext, scrollTop])

  if (existingContext) return <div {...props} />
  return (
    <ScrollRestorationContext.Provider value={scrollRef.current}>
      <div ref={scrollRef} {...props} />
    </ScrollRestorationContext.Provider>
  )
}
```

Comment (alex-adamant) [reply app/components/common/ScrollToTop.tsx]: So, give `setTimeout` a try.

**Pull request merged**